tests/Feature/PhotosTest.php

```diff
@@ -202,6 +198,56 @@ public function testUpload()
 		AccessControl::logout();
 	}
 
+	/**
+	 * Test live photo upload.
+	 *
+	 * @return void
+	 */
+	public function testLivePhotoUpload()
+	{
+		$photos_tests = new PhotosUnitTest($this);
+
+		AccessControl::log_as_id(0);
+		// MUST use exiftool to get live photo metadata
+		$init_config_value = Configs::get_value('has_exiftool');
+		Configs::set('has_exiftool', '1');
+
+		/*
+		 * Make a copy of the image because import deletes the file and we want to be
+		 * able to use the test on a local machine and not just in CI.
+		 */
+		copy('tests/Feature/train.jpg', 'public/uploads/import/train.jpg');
+		copy('tests/Feature/train.mov', 'public/uploads/import/train.mov');
+
+		$photo_file = new UploadedFile(
+			'public/uploads/import/train.jpg',
+			'train.jpg',
+			'image/jpeg',
+			null,
+			true
+		);
+
+		$video_file = new UploadedFile(
+			'public/uploads/import/train.mov',
+			'train.mov',
+			'video/quicktime',
+			null,
+			true
+		);
+
+		$photo_id = $photos_tests->upload($photo_file);
+		$video_id = $photos_tests->upload($video_file);
+
+		$photo = $this->asArray($photos_tests->get($photo_id, 'true'));
+
+		$this->assertEquals($photo_id, $video_id);
+		$this->assertEquals($photo->livePhotoContentID, 'E905E6C6-C747-4805-942F-9904A0281F02');
+		$this->assertStringEndsWith('.mov', $photo->livePhotoUrl);
```

Comment:
At first, this line puzzled me, because you could not invoke a property `->livePhotoUrl` on an array, if `$photo` was indeed an array as the name of the method `asArray` suggests. However, `asArray` does not return an array so the name of the method is actually not well chosen (see below).

However, the method `asArray` is not required at all. Just take the `TestReponse` which is returned by `$photos_test->get` and use the method `assertJson`. See `GeoDataTest` for examples how to test for particular JSON attrbiutes in a response.

Reply:
True, but here we only check two attributes, and one of them is not a direct equality. :)